massa-node/src/main.rs

```diff
@@ -177,15 +180,26 @@ async fn launch(
     })
     .expect("could not start selector worker");
 
-    // Create final state
-    let final_state = Arc::new(parking_lot::RwLock::new(
-        FinalState::new(
-            final_state_config,
-            Box::new(ledger),
-            selector_controller.clone(),
-        )
-        .expect("could not init final state"),
-    ));
+    // Create final state, from snapshot or from scratch
+    let final_state = match args.restart_from_snapshot_at_period {
+        Some(last_start_period) => Arc::new(parking_lot::RwLock::new(
+            FinalState::new_derived_from_snapshot(
+                final_state_config,
+                Box::new(ledger),
+                selector_controller.clone(),
+                last_start_period,
+            )
+            .expect("could not init final state"),
+        )),
+        None => Arc::new(parking_lot::RwLock::new(
+            FinalState::new(
+                final_state_config,
+                Box::new(ledger),
+                selector_controller.clone(),
+            )
+            .expect("could not init final state"),
+        )),
```

Comment:
consider putting the `Arc::new(parking_lot::RwLock::new(` around the match-statement

```rust
let final_state = Arc::new(parking_lot::RwLock::new(match ... {
    Some(last....) => FinalState::new_....(
    ....
     ) ,
    None => FinalState::new(
    ....
    )
}));
```